pkg/xray/resource_xray_license_policy.go

```diff
@@ -20,143 +21,173 @@ func resourceXrayLicensePolicyV2() *schema.Resource {
 												},
 												"active": {
-													Type:     schema.TypeBool,
-													Required: true,
+													Type:        schema.TypeBool,
+													Required:    true,
+													Description: "Whether or not to block download of artifacts that meet the artifact and severity `filters` for the associated `xray_watch` resource.",
 												},
 											},
 										},
 									},
 									"block_release_bundle_distribution": {
-										Type:     schema.TypeBool,
-										Optional: true,
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: "Blocks Release Bundle distribution to Edge nodes if a violation is found.",
 									},
 									"fail_build": {
-										Type:     schema.TypeBool,
-										Optional: true,
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: "Whether or not the related CI build should be marked as failed if a violation is triggered. This option is only available when the policy is applied to an `xray_watch` resource with a `type` of `builds`.",
 									},
 									"notify_deployer": {
-										Type:     schema.TypeBool,
-										Optional: true,
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: "Sends an email message to component deployer with details about the generated Violations.",
 									},
 									"notify_watch_recipients": {
-										Type:     schema.TypeBool,
-										Optional: true,
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: "Sends an email message to all configured recipients inside a specific watch with details about the generated Violations.",
 									},
 									"create_ticket_enabled": {
-										Type:     schema.TypeBool,
-										Optional: true,
-										Default:  false,
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Default:     false,
+										Description: "Create Jira Ticket for this Policy Violation. Requires configured Jira integration.",
 									},
 									"custom_severity": {
-										Type:     schema.TypeString,
-										Optional: true,
-										Default:  "High",
+										Type:             schema.TypeString,
+										Optional:         true,
+										Default:          "High",
+										Description:      "The severity of violation to be triggered if the `criteria` are met.",
+										ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"Critical", "High", "Medium", "Low"}, true)),
 									},
 									"build_failure_grace_period_in_days": {
-										Type:     schema.TypeInt,
-										Optional: true,
-										Default:  3,
+										Type:        schema.TypeInt,
+										Optional:    true,
+										Description: "Allow grace period for certain number of days. All violations will be ignored during this time. To be used only if `fail_build` is enabled.",
```

Comment:
We may want to validate for < 0.

Reply:
definitely validate - part of our/your job as a TF provider implementator is to guide your user